kafka-client/src/main/java/dev/responsive/kafka/internal/stores/PartitionedOperations.java

```diff
@@ -123,48 +123,62 @@ public static PartitionedOperations create(
         flushManager
     );
 
-    final CommitBuffer<Bytes, ?> buffer = CommitBuffer.from(
-        batchFlusher,
-        sessionClients,
-        changelog,
-        keySpec,
-        params.name(),
-        false,
-        config
-    );
-
-    final long restoreStartOffset = table.fetchOffset(changelog.partition());
-    final var registration = new ResponsiveStoreRegistration(
-        name.kafkaName(),
-        changelog,
-        restoreStartOffset == NO_COMMITTED_OFFSET
-            ? OptionalLong.empty()
-            : OptionalLong.of(restoreStartOffset),
-        buffer::flush,
-        streamThreadId()
-    );
-    storeRegistry.registerStore(registration);
-
-    final boolean migrationMode = ConfigUtils.responsiveMode(config) == ResponsiveMode.MIGRATE;
-    long startingTimestamp = -1;
-    final Optional<Duration> ttl = params.timeToLive();
-    if (migrationMode && ttl.isPresent()) {
-      startingTimestamp = Instant.now().minus(ttl.get()).toEpochMilli();
+    // these objects need to be cleaned up once they're created. If this method fails
+    // then we should make sure we do the cleanup here.
+    CommitBuffer<Bytes, ?> buffer = null;
+    ResponsiveStoreRegistration registration = null;
+    try {
```

Comment:
thoughts on making them implement `Closeable` and then using try-with-resources? that way we'll also get IDEA warnings if we forget to do it later

Reply:
ah yes, that makes sense to me!